### PR TITLE
Added generation script for RIF rules

### PR DESCRIPTION
**Pull request by situx**
As discussed in #70 there seems to be the need to create various combinations of RIF rules depending on the literal types and the functions involved.
I wrote a python script which generates the combinations automatically and expects the literal properties (geo:asGML, geo:asWKT) and the functions as input.
In this way, we can generate RIF rules automatically no matter how many literals we define in the future.

### CODE REVIEW

**Comment by nicholascar** [comment]
I did, of course, used a Python script to generate the RIF file I made back in PR #70 but I used a templating approach with the template in the standard and some header & footer templates to make a valid RIF document:

```python
# all
relations_functions = [
    # Simple Features Topological Relations
    "sfEquals",
    "sfDisjoint",
    "sfIntersects",
    "sfTouches",
    "sfWithin",
    "sfContains",
    "sfOverlaps",
    "sfCrosses",

    # Egenhofer Topological Relations
    "ehEquals",
    "ehDisjoint",
    "ehMeet",
    "ehOverlap",
    "ehCovers",
    "ehCoveredBy",
    "ehInside",
    "ehContains",

    # RCC8 Topological Relations
    "rcc8eq",
    "rcc8dc",
    "rcc8ec",
    "rcc8po",
    "rcc8tppi",
    "rcc8tpp",
    "rcc8ntpp",
    "rcc8ntppi",
]

geom_literals = [
    "gmlLiteral",
    "wktLiteral",
]

forall_groups = []
with open("template.rif") as f:
    temp = f.read()

    for rf in relations_functions:
        for lit in geom_literals:
            forall_groups.append(
                temp
                    .replace("ogc:relation", "geo:" + rf)
                    .replace("ogc:function", "geof:" + rf)
                    .replace("ogc:asGeomLiteral", "geo:" + lit)
            )

with open("rules.rifps", "w") as f2:
    f2.write(open("doc-top.rif").read())
    f2.write("\n\n".join(forall_groups))
    f2.write(open("doc-bottom.rif").read())
```

Template:
```
        # ogc:relation
        Forall ?f1 ?f2 ?g1 ?g2 ?g1Serial ?g2Serial (
            ?f1[ogc:relation->?f2] :-
            Or (
                # feature – feature rule
                And (
                    ?f1[geo:hasDefaultGeometry->?g1]
                    ?f2[geo:hasDefaultGeometry->?g2]
                    ?g1[ogc:asGeomLiteral->?g1Serial]
                    ?g2[ogc:asGeomLiteral->?g2Serial]
                    External(ogc:function (?g1Serial,?g2Serial))
                )
                # feature – geometry rule
                And (
                    ?f1[geo:hasDefaultGeometry->?g1]
                    ?g1[ogc:asGeomLiteral->?g1Serial]
                    ?f2[ogc:asGeomLiteral->?g2Serial]
                    External(ogc:function (?g1Serial,?g2Serial))
                )
                # geometry - feature rule
                And (
                    ?f2[geo:hasDefaultGeometry->?g2]
                    ?f1[ogc:asGeomLiteral->?g1Serial]
                    ?g2[ogc:asGeomLiteral->?g2Serial]
                    External(ogc:function (?g1Serial,?g2Serial))
                )
                # geometry - geometry rule
                And (
                    ?f1[ogc:asGeomLiteral->?g1Serial]
                    ?f2[ogc:asGeomLiteral->?g2Serial]
                    External(ogc:function (?g1Serial,?g2Serial))
                )
            )
        )
```
Header:
```
Document (
    Prefix (geo <http://www.opengis.net/ont/geosparql#>)
    Prefix (geof <http://www.opengis.net/def/function/geosparql/>)

    Group (

```
Footer:
```
    )
)
```


We can indeed put a script somewhere in the profiles - the profiling approach should allow us to include any relevant resources.

My script didn't cater for mixed input (GML & WKT of course).

### A way forward
Can we perhaps combine efforts here and use the updated logic in this code - the additional loop for the properties - but use a templating approach, rather than the `f.write()`? The template keeps the Python closer to the standard.

We could put the templates within the single Python file to keep the number of artefacts down.

@situx, would you be able to include the templating in this script?

**Comment by situx** [comment]
Everything integrated!

**Comment by nicholascar** [comment]
@situx I've done some stylistic-only updates to the script and its resultant rules.rifps file. I've also listed the file and the Python generator in the profile and move the rules and generator into the 1.1/ folder so they are specifically part of the 1.1 release. In PR #70 I've added the generator and updated rules.rifps to the 1.0/ folder.

I've update both the profile in RDF & HTML to include these files and also improved the profile a bit with descriptions for elements.

**Review by nicholascar — APPROVED** [review]

File: 1.1/rifgenerator.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    "rcc8ec",
+    "rcc8po",
+    "rcc8tppi",
+    "rcc8tpp",
+    "rcc8ntpp",
+    "rcc8ntppi",
+]
+
+geom_literals = [
+    "gmlLiteral",
```
Comment (mperry455) [inline 1.1/rifgenerator.py:38]: This should be the predicate instead of the datatype right? asGML, asWKT, etc.

**Review by mperry455 — APPROVED** [review]
Looks good. Thanks.

File: 1.1/rifgenerator.py (hunk 1 above)
Comment (nicholascar) [reply 1.1/rifgenerator.py]: Thanks for finding and fixing!

**Comment by nicholascar** [comment]
Good to merge @situx

**Pull request merged**